dsp/window/doc.go

```diff
@@ -0,0 +1,45 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package window provides a set of functions to perform the transformation
+// of sequence by different window functions.
+//
+// Window functions can be used to control spectral leakage parameters
+// when performing a Fourier transform on a signal of limited length.
+// See https://en.wikipedia.org/wiki/Window_function for more details.
+//
+// Spectral leakage parameters
+//
+// To ensure the selection criteria for a window function,
+// each function is documented by a set of spectral leakage parameters.
+// Here is a description.
+//
+// ΔF_0 -
+// Normalized width of the main AFC lobe at zero amplitude level.
+//
+// ΔF_0.5 -
+// Normalized width of the main AFC lobe at 0.5 amplitude level (-3 dB).
+//
+// K -
+// For a rectangular window, ΔF_0 has the smallest value and is equal to 2.
+// K Shows how many times the normalized width ΔF_0
+// of the main lobe by the zero level of the specified window
+// is wider than the rectangular window.
+// Depending on the parameter, the windows are divided into
+// high resolution windows (K≤3) and low resolution windows (K>3).
+//
+// ɣ_max -
+// The maximum level of side lobes.
+// If the maximum level of side lobes divided by maximum level of main lobe
+// is a g, the ɣ_max is a g expressed in logarithmic scale:
+// ɣ_max=20*log_10(g), [dB].
+//
+// β -
+// The reduction coefficient.
+// The meaning of the reduction coefficient
+// is that the amplitudes of all spectral components
+// after multiplying by the window function are reduced by b times
+// compared to the rectangular window. The reduction coefficient β
+// is a b expressed in logarithmic scale:  β=20*log_10(b), [dB].
```

Comment:
> // The characteristic changes associated with each window function may
> // be described using a set of spectral leakage parameters; β, ΔF_0, ΔF_0.5,
> // K and ɣ_max.
My be it should be the colon instead of the semicolon before β?

Reply:
I think, that `β_w = 20 log10(A_w / A_rect) dB` is a little bit confusing when we talk about β.
Here is two options:

- s/β/β_w/g.
- s/β_w = 20 log10(A_w / A_rect) dB/ β = 20 log10(A_window / A_rectangular) dB, where A is a constant component of the window function's Fourier image./

Reply:
// The β, attenuation coefficient of a window is the ratio of the amplitude
// of the main lobe of the spectrum resulting from use of the window compared
// to that produced using the rectangular window, expressed
// in a logarithmic scale.

here is my mistake. It's not about main lobe amplitude. The next is closer:

// The β, attenuation coefficient of a window is the ratio 
// of the constant component of the Fourier image of the window 
// compared to that produced using the rectangular window, 
// expressed in a logarithmic scale.

Reply:
I think, that `K_w = ΔF_0_w/ΔF_0_rect` is a little bit confusing when we talk about K.
Here is two options:

s/K/K_w/g.
s/K_w = ΔF_0_w/ΔF_0_rect/ K = ΔF_0_window/ΔF_0_rectangular/.

Reply:
may be it should be mentioned that:
// The ɣ_max parameter is the maximum level of the side lobes in decibels
// on the normalized Fourier image of the window function.

Reply:
For the K and β, the absence of the subscript in English is fine since it's talking about the value for a window, which is synonymous with K_w and β_w. I think this is fine.

I've incorporated your suggestions with some stylistic changes in the sections on β and ɣ_max.

```
// Spectral leakage parameters
//
// Application of window functions to an input will result in changes
// to the frequency content of the signal in an effect called spectral
// leakage. See https://en.wikipedia.org/wiki/Spectral_leakage.
//
// The characteristic changes associated with each window function may
// be described using a set of spectral leakage parameters; β, ΔF_0, ΔF_0.5,
// K and ɣ_max.
//
// The β, attenuation, coefficient of a window is the ratio of the
// constant component of the spectrum resulting from use of the window
// compared to that produced using the rectangular window, expressed in
// a logarithmic scale.
//  β_w = 20 log10(A_w / A_rect) dB
//
// The ΔF_0 parameter describes the normalized width of the main lobe of
// the frequency spectrum at zero amplitude.
//
// The ΔF_0.5 parameter describes the normalized width of the main lobe of
// the frequency spectrum at -3 dB (half maximum amplitude).
//
// The K parameter describes the relative width of the main lobe of the
// frequency spectrum produced by the window compared with the rectangular
// window. The rectangular window has the lowest ΔF_0 at a value of 2.
//  K_w = ΔF_0_w/ΔF_0_rect.
// The value of K divides windows into high resolution windows (K≤3) and
// low resolution windows (K>3).
//
// The ɣ_max parameter is the maximum level of the side lobes of the
// normalized spectrum, in decibels.
```

Also, if you want, add this text as a comment below the package declaration.

```
// The article at http://www.dsplib.ru/content/win/win.html is the origin
// of much of the information used in this package. It's in Russian, but
// Google translate does a pretty good job with it.
```

I think it's worth adding, but unless it's the seminal article, a foreign language article is probably out of place for user-facing documentation.